From 51a7de4f2abb2f40a9a685019bb84b1e840a9abe Mon Sep 17 00:00:00 2001
From: Kevin Chen <kevin_chen@aspeedtech.com>
Date: Thu, 21 Aug 2025 16:25:47 +0800
Subject: [PATCH 10/32] misc: edaf_bridge: Update compatible strings and add
 configuration support

Updates the compatible strings for eDAF bridges to distinguish between
eSPI0 and eSPI1 instances. Adds platform-specific configuration and
handling for enabling the eDAF bridge based on the platform type.

Change-Id: If92df0898d224b3fb9116158d38d106bc6357af8
Signed-off-by: Kevin Chen <kevin_chen@aspeedtech.com>
(cherry picked from commit b2e2eff7598278b204200ccb533c6d49ea094903)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/riscv/dts/ast2700.dtsi       |  4 ++--
 drivers/misc/aspeed_edaf_bridge.c | 39 ++++++++++++++++++++-----------
 2 files changed, 27 insertions(+), 16 deletions(-)

diff --git a/arch/riscv/dts/ast2700.dtsi b/arch/riscv/dts/ast2700.dtsi
index 791b2435f8..474e9f25c7 100644
--- a/arch/riscv/dts/ast2700.dtsi
+++ b/arch/riscv/dts/ast2700.dtsi
@@ -275,14 +275,14 @@
 		};
 
 		edaf_bridge0: edaf-bridge@14c38000 {
-			compatible = "aspeed,ast2700-edaf-bridge";
+			compatible = "aspeed,ast2700-edaf-bridge0";
 			reg = <0x14c38000 0x200>;
 			aspeed,scu1 = <&syscon1>;
 			status = "disabled";
 		};
 
 		edaf_bridge1: edaf-bridge@14c38200 {
-			compatible = "aspeed,ast2700-edaf-bridge";
+			compatible = "aspeed,ast2700-edaf-bridge1";
 			reg = <0x14c38200 0x200>;
 			aspeed,scu1 = <&syscon1>;
 			status = "disabled";
diff --git a/drivers/misc/aspeed_edaf_bridge.c b/drivers/misc/aspeed_edaf_bridge.c
index bae0f7bf42..d26f69115c 100644
--- a/drivers/misc/aspeed_edaf_bridge.c
+++ b/drivers/misc/aspeed_edaf_bridge.c
@@ -13,6 +13,10 @@
 #include <linux/ioport.h>
 #include <reset.h>
 
+#define UNKNOWN 0
+#define EDAF_BDGE_OVER_ESPI0 1
+#define EDAF_BDGE_OVER_ESPI1 2
+
 #define SCU1_HOST_CONF_1		0xa00
 #define SCU1_HOST_CONF_2		0xa20
 #define   SCU1_HOST_CONF_EDAF_EN	BIT(10)
@@ -59,6 +63,7 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 	ofnode node, scu1_node, mem_node;
 	uint64_t cbase, mbase, espibase;
 	uint32_t scu, edaf, espi, phandle;
+	uint32_t plat = (unsigned long)dev_get_driver_data(dev);
 	struct resource res;
 	int rc;
 
@@ -92,21 +97,26 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 		return -ENODEV;
 	}
 
-	scu = readl(scu_regs + SCU1_HOST_CONF_1);
-	if (scu & SCU1_HOST_CONF_LPC_EN) {
-		printf("Host Configuration1: eSPI0 eDAF is not valid\n");
-		return -EINVAL;
-	}
-	scu |= SCU1_HOST_CONF_EDAF_EN;
-	writel(scu, scu_regs + SCU1_HOST_CONF_1);
-
-	scu = readl(scu_regs + SCU1_HOST_CONF_2);
-	if (scu & SCU1_HOST_CONF_LPC_EN) {
-		printf("Host Configuration2: eSPI1 eDAF is not valid\n");
+	if (plat == EDAF_BDGE_OVER_ESPI0) {
+		scu = readl(scu_regs + SCU1_HOST_CONF_1);
+		if (scu & SCU1_HOST_CONF_LPC_EN) {
+			printf("Host Configuration1: eSPI0 eDAF is not valid\n");
+			return -EINVAL;
+		}
+		scu |= SCU1_HOST_CONF_EDAF_EN;
+		writel(scu, scu_regs + SCU1_HOST_CONF_1);
+	} else if (plat == EDAF_BDGE_OVER_ESPI1) {
+		scu = readl(scu_regs + SCU1_HOST_CONF_2);
+		if (scu & SCU1_HOST_CONF_LPC_EN) {
+			printf("Host Configuration2: eSPI1 eDAF is not valid\n");
+			return -EINVAL;
+		}
+		scu |= SCU1_HOST_CONF_EDAF_EN;
+		writel(scu, scu_regs + SCU1_HOST_CONF_2);
+	} else {
+		printf("Unknown platform for eDAF bridge\n");
 		return -EINVAL;
 	}
-	scu |= SCU1_HOST_CONF_EDAF_EN;
-	writel(scu, scu_regs + SCU1_HOST_CONF_2);
 
 	edaf = readl(edaf_bridge_regs + EDAF_BDGE_CFG);
 	if (ofnode_read_bool(node, "edaf-ddr-mode")) {
@@ -201,7 +211,8 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 }
 
 static const struct udevice_id aspeed_edaf_bridge_ids[] = {
-	{ .compatible = "aspeed,ast2700-edaf-bridge" },
+	{ .compatible = "aspeed,ast2700-edaf-bridge0", .data = (unsigned long)EDAF_BDGE_OVER_ESPI0 },
+	{ .compatible = "aspeed,ast2700-edaf-bridge1", .data = (unsigned long)EDAF_BDGE_OVER_ESPI1 },
 	{ }
 };
 
-- 
2.34.1

